Examples/Detectors/TelescopeDetector/include/ActsExamples/TelescopeDetector/BuildTelescopeDetector.hpp

```diff
@@ -0,0 +1,155 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Definitions/Units.hpp"
+#include "Acts/Geometry/CuboidVolumeBounds.hpp"
+#include "Acts/Geometry/LayerArrayCreator.hpp"
+#include "Acts/Geometry/LayerCreator.hpp"
+#include "Acts/Geometry/PlaneLayer.hpp"
+#include "Acts/Geometry/SurfaceArrayCreator.hpp"
+#include "Acts/Geometry/TrackingGeometry.hpp"
+#include "Acts/Geometry/TrackingGeometryBuilder.hpp"
+#include "Acts/Geometry/TrackingVolume.hpp"
+#include "Acts/Geometry/TrackingVolumeArrayCreator.hpp"
+#include "Acts/Material/HomogeneousSurfaceMaterial.hpp"
+#include "Acts/Material/Material.hpp"
+#include "Acts/Material/ProtoSurfaceMaterial.hpp"
+#include "Acts/Surfaces/PlaneSurface.hpp"
+#include "Acts/Surfaces/RectangleBounds.hpp"
+#include "Acts/Surfaces/SurfaceArray.hpp"
+#include "Acts/Utilities/BinUtility.hpp"
+#include "Acts/Utilities/Logger.hpp"
+#include "ActsExamples/TelescopeDetector/TelescopeDetectorElement.hpp"
+
+#include <array>
+#include <cmath>
+#include <iostream>
+#include <list>
+#include <memory>
+#include <vector>
+
+namespace Acts {
+class TrackingGeometry;
+}
+
+namespace ActsExamples {
+namespace Telescope {
+
+/// Global method to build the telescope tracking geometry
+///
+/// @tparam detector_element_t is the actual type of the detector
+/// element, each derivative of a TelescopeDetectorElement can be used
+///
+/// @param gctx is the detector element dependent geometry context
+/// @param detectorStore is the store for the detector element
+/// @param positions is the offset w of different layers in the longitudinal
+/// direction
+/// @param offsets is the offset (u, v) of the layers in the transverse plane
+/// @param pSize is the plane size
+/// @param thickness is the material thickness of each layer
+/// @param binValue indicates which axis the planes normals are parallel to
+template <typename detector_element_t>
+std::unique_ptr<const Acts::TrackingGeometry> buildDetector(
```

Comment:
Does this need to be templated on the detector element? This should always be the `TelescopeDetectorElement` and the function can be implemented within a `.cpp` file instead, right?

Reply:
No, it doesn't have to. Template not used now.